src/Lykke.Job.BlockchainCashinDetector.Core/Domain/CashinAggregate.cs

```diff
@@ -201,9 +217,13 @@ public bool OnClientOperationStartRegistered()
 
         public bool OnOperationCompleted(string transactionHash, long transactionBlock, decimal transactionAmount, decimal fee)
         {
-            if (!SwitchState(
-                new[] {CashinState.ClientOperationStartIsRegistered, CashinState.EnrolledToMatchingEngine},
-                CashinState.OperationIsFinished))
+            var expectedStates = new[]
+            {
+                CashinState.ClientOperationStartIsRegistered,
+                CashinState.EnrolledBalanceIncreased
+            };
+
+            if (!SwitchState(expectedStates, CashinState.EnrolledBalanceReset))
```

Comment:
It looks strange, that on operation completed, aggregate switches to the EnrolledBalanceReset state